src/lib/operators/table_scan.cpp

```diff
@@ -43,7 +43,21 @@ namespace opossum {
 TableScan::TableScan(const std::shared_ptr<const AbstractOperator>& in,
                      const std::shared_ptr<AbstractExpression>& predicate)
     : AbstractReadOnlyOperator{OperatorType::TableScan, in, nullptr, std::make_unique<PerformanceData>()},
-      _predicate(predicate) {}
+      _predicate(predicate) {
+  /**
+   * Register as a consumer for all subplans of uncorrelated subqueries
+   */
+  auto pqp_subquery_expressions = collect_pqp_subquery_expressions(predicate);
+  for (const auto& subquery_expression : pqp_subquery_expressions) {
+    // We do not register for the subplans of correlated subqueries because they are templated and cannot be
+    // executed without concrete parameters. Thus, there is no option for result sharing at this point.
+    if (subquery_expression->is_correlated()) continue;
+
+    // Register & Store a pointer for easy deregistration later on.
+    subquery_expression->pqp->register_consumer();
+    _uncorrelated_subquery_expressions.push_back(subquery_expression);
+  }
+}
```

Comment:
Basically the same two or three points that I mentioned in projection.cpp apply here too.